src/controllers/user/routes.ts

```diff
@@ -3,14 +3,16 @@ import userControler from './controller';
 import validationHandler from '../../libs/validationHandler';
  import config from './Validation';
 import authmiddleware from '../../libs/routes/authmiddleware';
+import database from '../../libs/database';
+import validation from './Validation';
+import { permissions } from '../../libs/constants';
 console.log(authmiddleware);
 const userRouter = Router();
 userRouter.route('/')
-
 .get(authmiddleware('getUser', 'read'), validationHandler ( config.get ) , userControler.get)
 .post(authmiddleware('getUser', 'read'), validationHandler ( config.create) , userControler.create)
-.put(authmiddleware('getUser', 'read'), validationHandler ( config.update) ,  userControler.update)
-.delete(authmiddleware('getUser', 'read'), validationHandler ( config.delete) , userControler.delete);
+.put(authmiddleware('getUser', ' read'), validationHandler ( config.update) ,  userControler.update)
+userRouter.route('/:id').delete(authmiddleware('getUser', 'read'), validationHandler ( config.delete) , userControler.delete);
```

Comment:
Do not pass permission type 'read' for all routes in auth middleware use 'read' for GET, 'write' for create and update (PUT ,POST) , and 'delete' for DELETE